.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,49 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master, release/* ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '23 17 * * 3'
+
+permissions: read-all
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
```

Comment:
do we want this to run on other platforms too? I guess if it's running `make` then our hands are kinda tied 🤷

Reply:
I don't know how this works exactly, but as a code-scanner it shouldn't much matter.  I wonder if it extracts paths from the debugging symbols.